File: api/src/endpoint/task/poll_task.rs
use crate::{endpoint::Endpoint, resource::TaskResource};

pub struct PollTask;

impl Endpoint for PollTask {
    const PATH: &'static str = "/ci/task/poll";
    const METHOD: http::Method = http::Method::GET;

    type Request = PollTaskRequest;
    type Response = PollTaskResponse;
}

pub type PollTaskRequest = ();

pub type PollTaskResponse = TaskResource;

File: gitdot-api-ts/src/endpoint/task/poll-task.ts
import { z } from "zod";

import { PollTaskResource } from "../../resource";

export const PollTaskRequest = z.object({});
export type PollTaskRequest = z.infer<typeof PollTaskRequest>;

export const PollTask = {
  path: "/ci/task/poll",
  method: "GET",
  request: PollTaskRequest,
  response: PollTaskResource.nullable(),
} as const;
export type PollTask = typeof PollTask;
